include/xtensor/xview.hpp

```diff
@@ -609,9 +611,13 @@ namespace xt
     template <class CT, class... S>
     template <class T>
     inline auto xview<CT, S...>::strides() const ->
-        std::enable_if_t<has_raw_data_interface<T>::value, strides_type>
+        std::enable_if_t<has_raw_data_interface<T>::value, strides_type&>
     {
-        strides_type strides = xtl::make_sequence<strides_type>(m_e.dimension() - integral_count<S...>(), 0);
+        if (m_strides_computed)
+        {
+            return m_strides;
+        }
```

Comment:
I would prefer something like:
```cpp
if(!m_strides_computed)
{
    m_strides = compute_strides();
    m_strides_computed = true;
}
return m_strides;
```
for the following reasons:
- you avoid mutiple return points
- all the caching logic is done in the same place, and `compute_strides` is now a stateless function (that's more flexible for future refactoring / cache policy changes etc).